### PR TITLE
Updated the build document

### PR DESCRIPTION
**Pull request by ktabata**
* Added the download URL for Intel Mac
* Added the explanation for `make-dmg.sh`
* Added the explanation for `do-release.sh`

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -26,16 +26,25 @@ This method uses a cross compiler to build a Windows binary.
...
 
-* On macOS 12, install Xcode 13.
+* On macOS 13, install Xcode 14.
 * From the terminal, enter the `build/macos` directory.
-   * Run `./build-libs.sh` to build libraries.
+   * On Apple Silicon Mac:
+      * Run `./build-libs.sh` to build the libraries.
+   * On Intel Mac:
+      * You can't build a `Universal Binary` library for `libpng`, so use the prebuilt libraries.
+      * Download these from `https://suika2.com/dl/libroot-mac.tar.gz` and extract it.
```
Comment (KyouBrayden) [inline build/README.md:35]: 'it' –> 'them'

File: build/README.md [hunk 2]
```diff
@@ -26,16 +26,25 @@ This method uses a cross compiler to build a Windows binary.
...
 * In Xcode, open `build/macos`.
    * Set your development team.
    * Build the project.
    * Archive then project.
    * Notarize the application by `Distribute App` button.
-   * Press `Export Notarized App` to export the app to `suika2` folder.
+   * Press `Export Notarized App` to export the app to `build/macos` folder.
```
Comment (KyouBrayden) [inline build/README.md:41]: 'to `build/macos`' –> 'to the `build/macos`'

File: build/README.md (hunk 1 above)
Comment (ktabata) [reply build/README.md:35]: Fixed.

File: build/README.md (hunk 2 above)
Comment (ktabata) [reply build/README.md:41]: Fixed.

**Review by KyouBrayden — APPROVED** [review]
LGTM!
Thank you!

**Pull request merged**